cmd/wharf/run.go

```diff
@@ -85,13 +91,23 @@ https://iver-wharf.github.io/#/usage-wharfyml/`,
 			return err
 		}
 
-		ctx := context.Background()
+		ctx, cancel := context.WithCancel(context.Background())
 		if runFlags.serve {
 			var server workerserver.Server
 			ctx, server = startWorkerServerWithCancel(ctx, store)
 			defer server.Close()
 		}
 
+		go handleCancelSignals(func() {
+			go func() {
+				time.AfterFunc(cancelGracePeriod, func() {
+					log.Warn().Message("Failed to cancel within grace period. Force quitting now.")
+					os.Exit(3)
```

Comment:
Suggest extracting these exit codes to a `const` declaration at the top so that we can more easily keep track of the exit codes we use.

Then also extract the exit code used in `execute()` inside `root.go`

Something like:

```go
const (
    exitCodeError           = 1
    exitCodeCancelForceQuit = 2
    exitCodeCancelTimeout   = 3
)
```